### PR TITLE
Remove warnings, disable warnings from fftw3 library.

### PR DESCRIPTION
**Pull request by drslebedev**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -65,12 +65,13 @@ target_include_directories(pmtv INTERFACE ${pmt_SOURCE_DIR}/include/)
 
 # FFTW3 is build 2 times for float and double precisions
 SET(FFTW_PREFIX ${CMAKE_CURRENT_BINARY_DIR}/fftw)
+SET(FFTW_CFLAGS "${CFLAGS} -w")
 if (EMSCRIPTEN)
     SET(FFTW_CONFIG cd ${FFTW_PREFIX}/src/ && emconfigure ./configure --quiet --disable-fortran --prefix=${FFTW_PREFIX}/install)
-    SET(FFTW_BUILD emmake make -j && emmake make install && emmake make clean)
+    SET(FFTW_BUILD emmake make -j CFLAGS=${FFTW_CFLAGS} && emmake make install && emmake make clean)
```
Comment (wirew0rm) [inline CMakeLists.txt:71]: Looking at the build output I think we might also want to pass `--silent` to make to suppress the echo of the compile commands it is running? Right now the CI build step mostly shows the output of make for fftw (and that even two times).

**Review by wirew0rm — APPROVED** [review]
Looks good, thanks for cleaning up the build. I added 2 suggestions, feel free to incorporate if you agree and merge this.

File: test/qa_fft.cpp [hunk 2]
```diff
@@ -99,8 +99,7 @@ const boost::ut::suite fftTests = [] {
             std::ignore = fft1.settings().set({ { "outputInDb", t.outputInDb } });
             std::ignore = fft1.settings().apply_staged_parameters();
             const auto signal{ generateSinSample<T>(t.N, t.sampleRate, t.frequency, t.amplitude) };
-
-            fft1.inputHistory.push_back_bulk(signal.begin(), signal.end());
+            for (const auto s : signal) fft1.inputHistory.push_back(static_cast<typename fft<T>::InHistoryType>(s));
```
[the remark this answers is not in the document]
Comment (drslebedev) [reply test/qa_fft.cpp:102]: Thanks for pointing this out. I included braces.

File: CMakeLists.txt (hunk 1 above)
Comment (drslebedev) [reply CMakeLists.txt:71]: I included `--silent`. There is still some output form `fftw`, I'm not sure if we can control it.

**Pull request merged**